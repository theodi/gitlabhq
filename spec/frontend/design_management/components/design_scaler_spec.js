import { shallowMount } from '@vue/test-utils';
import DesignScaler from '~/design_management/components/design_scaler.vue';

describe('Design management design scaler component', () => {
  let wrapper;

  function createComponent(propsData, data = {}) {
    wrapper = shallowMount(DesignScaler, {
      propsData,
    });
    wrapper.setData(data);
  }

  afterEach(() => {
    wrapper.destroy();
  });

  const getButton = type => {
    const buttonTypeOrder = ['minus', 'reset', 'plus'];
    const buttons = wrapper.findAll('button');
    return buttons.at(buttonTypeOrder.indexOf(type));
  };

  it('emits @scale event when "plus" button clicked', () => {
    createComponent();

    getButton('plus').trigger('click');
    expect(wrapper.emitted('scale')).toEqual([[1.2]]);
  });

  it('emits @scale event when "reset" button clicked (scale > 1)', () => {
    createComponent({}, { scale: 1.6 });
    return wrapper.vm.$nextTick().then(() => {
      getButton('reset').trigger('click');
      expect(wrapper.emitted('scale')).toEqual([[1]]);
    });
  });

  it('emits @scale event when "minus" button clicked (scale > 1)', () => {
    createComponent({}, { scale: 1.6 });

    return wrapper.vm.$nextTick().then(() => {
      getButton('minus').trigger('click');
      expect(wrapper.emitted('scale')).toEqual([[1.4]]);
    });
  });

  it('minus and reset buttons are disabled when scale === 1', () => {
    createComponent();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('minus and reset buttons are enabled when scale > 1', () => {
    createComponent({}, { scale: 1.2 });
    return wrapper.vm.$nextTick().then(() => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  it('plus button is disabled when scale === 2', () => {
    createComponent({}, { scale: 2 });
    return wrapper.vm.$nextTick().then(() => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
