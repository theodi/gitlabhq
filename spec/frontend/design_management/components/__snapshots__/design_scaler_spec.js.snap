// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design management design scaler component minus and reset buttons are disabled when scale === 1 1`] = `
<div
  class="design-scaler btn-group"
  role="group"
>
  <button
    class="btn"
    disabled="disabled"
  >
    <span
      class="gl-display-flex gl-justify-content-center gl-align-items-center gl-icon s16"
    >
      
      –
    
    </span>
  </button>
   
  <button
    class="btn"
    disabled="disabled"
  >
    <gl-icon-stub
      name="redo"
      size="16"
    />
  </button>
   
  <button
    class="btn"
  >
    <gl-icon-stub
      name="plus"
      size="16"
    />
  </button>
</div>
`;

exports[`Design management design scaler component minus and reset buttons are enabled when scale > 1 1`] = `
<div
  class="design-scaler btn-group"
  role="group"
>
  <button
    class="btn"
  >
    <span
      class="gl-display-flex gl-justify-content-center gl-align-items-center gl-icon s16"
    >
      
      –
    
    </span>
  </button>
   
  <button
    class="btn"
  >
    <gl-icon-stub
      name="redo"
      size="16"
    />
  </button>
   
  <button
    class="btn"
  >
    <gl-icon-stub
      name="plus"
      size="16"
    />
  </button>
</div>
`;

exports[`Design management design scaler component plus button is disabled when scale === 2 1`] = `
<div
  class="design-scaler btn-group"
  role="group"
>
  <button
    class="btn"
  >
    <span
      class="gl-display-flex gl-justify-content-center gl-align-items-center gl-icon s16"
    >
      
      –
    
    </span>
  </button>
   
  <button
    class="btn"
  >
    <gl-icon-stub
      name="redo"
      size="16"
    />
  </button>
   
  <button
    class="btn"
    disabled="disabled"
  >
    <gl-icon-stub
      name="plus"
      size="16"
    />
  </button>
</div>
`;
